.github/workflows/validate-templates.yaml

```
@@ -0,0 +1,75 @@
+      - '.github/workflows/validate-templates.yml'
+  workflow_dispatch:
+
+jobs:
+  validate:
+    name: Validate Template Files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v4
```

actionlint (1.7.7)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)